control-plane/csi-driver/controller/src/pvwatcher.rs

```diff
@@ -0,0 +1,145 @@
+use anyhow::Result;
+use futures::TryStreamExt;
+use k8s_openapi::api::core::v1::PersistentVolume;
+
+use kube::{
+    api::{Api, ListParams},
+    runtime::{watcher, WatchStreamExt},
+    Client,
+};
+use tracing::{debug, error, info};
+use uuid::Uuid;
+
+use crate::IoEngineApiClient;
+
+const MAYASTOR_PLUGIN: &str = "io.openebs.csi-mayastor";
+
+pub struct PvGarbageCollector {}
+
+impl PvGarbageCollector {
+    pub async fn start_watcher() -> Result<(), watcher::Error> {
+        info!("Starting PV Garbage Collector");
+        debug!("Handling if any missed events");
+        tokio::spawn(async {
+            handle_missed_events().await;
+        });
+        let client = Client::try_default().await.unwrap();
```

Comment:
please handle the error instead of unwrap.

Reply:
There are some places where i feel we will always get Some(). I think this is one of them. We will get inclusterconfig.

Reply:
That may be true, but IMHO it's always better to handle errors properly.